packages/SwingSet/src/kernel/kernel.js

```diff
@@ -1296,6 +1328,11 @@ export default function buildKernel(
         policyInput = ['crank', {}];
       }
     }
+    if (message.type === 'cleanup-terminated-vat') {
+      const { cleanups } = crankResults;
+      assert(cleanups !== undefined);
+      policyInput = ['cleanup', { cleanups }];
+    }
```

Comment:
This feels like it should be checked first.
```js
    const vatID = crankResults.didDelivery;
    if (message.type === 'cleanup-terminated-vat') {
      const { cleanups } = crankResults;
      assert(cleanups !== undefined);
      policyInput = ['cleanup', { cleanups }];
    } else if (vatID) {
      const policyInputTag = message.type === 'create-vat' ? 'create-vat' : 'crank';
      policyInput = [policyInputTag, {}];
    }
```

Reply:
yeah, although I don't think I want to assign `vatID` this far out (i.e. leave it as `else if (crankResults.didDelivery) {`). Looking later in this function to find out where `vatID` is assigned/extracted/used, I see that we've got (probably) redundant copies in `.measureDirt` and `.terminate`. The intention of `crankResults` was to be a set of instructions for post-delivery code to act upon, but I don't think there's any way for a delivery to vat1 to e.g. add dirt to vat2.

So I think we have a cleanup pass to make, to consolidate those multiple `.vatID` properties, but also to simplify the `crankResults` language.

Reply:
so:
```js

    if (message.type === 'cleanup-terminated-vat') {
      const { cleanups } = crankResults;
      assert(cleanups !== undefined);
      policyInput = ['cleanup', { cleanups }];
    } else if (crankResults.didDelivery) {
      const tag = message.type === 'create-vat' ? 'create-vat' : 'crank';
      policyInput = [tag, {}];
    }
```